radiometric_normalization/gimage.py

```diff
@@ -0,0 +1,120 @@
+'''
+Copyright 2015 Planet Labs, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+'''
+import logging
+from collections import namedtuple
+
+import numpy
+from osgeo import gdal, gdal_array
+
+# In-memory geospatial image
+# Bands: list of uint16 numpy arrays, each holding a band of data
+# Alpha: uint16 numpy array holding the alpha information
+# Metadata: dict containing georeferencing information
+# (geotransform, projection, rpc)
+GImage = namedtuple('GImage', 'bands, alpha, metadata')
+
+
+def save(gimage, filename):
```

Comment:
This is a great idea. I implemented a tweaked version to accommodate the gdal paradigm:

```
def save(gimage, filename):
    gdal_ds = create_ds(gimage, filename)
    save_to_ds(gimage, gdal_ds)
``` 

`create_ds` creates the output gdal dataset (similar to file_object) and `save_to_ds` writes the gimage to the gdal_ds. One gotcha here is that gdal_ds is a pointer and it must be set to None for the file to be generated. Example:

```
output_ds = gdal.GetDriverByName('GTiff').Create(
            output_file, 2, 2, 2, gdal.GDT_UInt16,
            options=['PHOTOMETRIC=RGB', 'ALPHA=YES'])

gimage.save_to_ds(test_gimage, output_ds)
output_ds = None
``